### PR TITLE
WIP:Feature40 791/835/950/989

### PR DESCRIPTION
**Pull request by varsha-22**
Tasks done:
Task I
1. Setup the Material UI theme 
2. Create a theme.js file in “src” folder.
3. Setup the theme in “App.js” using above created theme.js file.
4. Now create a component named as “Math” that will accept following props:
* first - a number and required
* second - a number and required
* operator - must be any of ‘+’, ‘-’, ‘/’, ‘*’
* children - optional and must contain function that will be used to render the template provided by the user, and will accept the object with keys, first, second, operator, and result. This will used to provide the template to the Math component as children.
5. Please note in case of “/” operator if second value is 0 then result must be “Infinity”
6. If anyhow operator must not be in the provide list then result must be “Invalid Operator”.
7. Default template is:
- 7 + 4 = 11
- 7 - 3 = 4
- 7 / 0 = Infinity
- 7 ^ 8 = Invalid Operation.
8. Using children we can provide various templates to the component and can print the result in any format;
- Sum of 7 and 4 is 11
- When we add 3 with 4 then we will get 7 as result.
- and so on.
9. Show the various usage of Math Component in “ChildrenDemo” component. It will be similar to another Demo component.
10. One of the user-defined templates must use the “Typography” component from “Material UI”

Task II
Create a Trainee Component similar to other page component.
2) Inside Trainee Component create a folder named as components and create a new component AddDialog component
3) AddDialog compoent will accept following props:
* open - boolean, will open the dialog if true.
* onClose - handler to close the dialog, passed from TraineeComponent
* onSubmit - handler to accept the dialog state for user: name, email, password
4) AddDialog will show information as show in the GIF
5) Add all the validations as shown in the GIF, similar to InputDemo component.
6) All the components used in the AddDialog must be from the Material UI.
7) TraineeComponent - Will hold the state of the Dialog, and update the state as per handler Request.
8) TraineeComponent - Will contain the button, and clicking on the button will open the dialog inside “AddDialog” component.

Task III
1) Create a Trainee Component similar to other page component.
2) Inside Trainee Component create a folder named as components and create a new component AddDialog component
3) AddDialog compoent will accept following props:
* open - boolean, will open the dialog if true.
* onClose - handler to close the dialog, passed from TraineeComponent
* onSubmit - handler to accept the dialog state for user: name, email, password
4) AddDialog will show information as show in the GIF
5) Add all the validations as shown in the GIF, similar to InputDemo component.
6) All the components used in the AddDialog must be from the Material UI.
7) TraineeComponent - Will hold the state of the Dialog, and update the state as per handler Request.
8) TraineeComponent - Will contain the button, and clicking on the button will open the dialog inside “AddDialog” component.

task IV
1. We need to setup the routes for all the page components that we have created 
2. Create a NotFound Component, and this will be rendered, if any of the route doesn’t match.
3. Create two types of routes i.e. AuthRoute and PrivateRoute, that will be used to render AuthLayout and PrivateLayout, along with the component.
4. Follow this for setting up the routes: https://gist.github.com/madcoda/8218dae875c69871d2ced69138042de0
5. Folder structure must be structured like shown as “folder.gif”
6. Move Navbar component to layouts folder as shown in “folder.gif”
7. Update App.js file to setup various routes using above custom created routes i.e. AuthRoute, PrivateRoute.
8. Create a Footer Component that will be used by AuthLayout as shown in the GIF. (Successive Technologies)

### CODE REVIEW

File: src/components/Math/Math.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            second: 0,
+            operator: null,
+            result:null
+        }
+    }
+    // const { first, second, operator } = this.state;
+    handleResultButton = (event) => {
+        event.preventDefault();
+    // eslint-disable-next-line no-unused-vars
+        let result = this.state.result;
```
Comment (pinku4444) [inline src/components/Math/Math.jsx:20]: Use Destructuring concept and remove all console.log

File: src/components/RadioGroup/RadioGroups.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// eslint-disable-next-line no-unused-vars
+import React, { Component } from 'react'
+// eslint-disable-next-line no-unused-vars
+import { RadioGroup, Radio } from 'react-radio-group';
+
+class RadioGroups extends Component {
+render() {
+    console.log(this.props);
```
Comment (pinku4444) [inline src/components/RadioGroup/RadioGroups.jsx:8]: Remove console.log

File: src/images/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+ const images = {
+    cloud : require('./cloud.jpg'),
```
Comment (pinku4444) [inline src/images/index.js:2]: Use import

File: src/layouts/AuthLayout/AuthLayout.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line no-unused-vars
+import React, { Component } from 'react';  
+import { Route } from 'react-router-dom';  
+  
+const LoginLayout = ({ children }) => (
```
Comment (pinku4444) [inline src/layouts/AuthLayout/AuthLayout.jsx:5]: create separate file for this component.

File: src/layouts/PrivateLayout/PrivateLayout.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-unused-vars */
+import React, { Component } from 'react';  
+import { Route } from 'react-router-dom';  
+  
+const DashboardLayout = ({children, ...rest}) => {
```
Comment (pinku4444) [inline src/layouts/PrivateLayout/PrivateLayout.jsx:5]: Create separate file for this component

File: src/layouts/components/Navbar/NavBar.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  },
+}));
+
+export default function NavBar(props) {
+  const classes = useStyles();
+  console.log('props',props);
+  return (
+    <div className={classes.root}>
+      <AppBar position="static">
+        <Toolbar>
```
Comment (pinku4444) [inline src/layouts/components/Navbar/NavBar.jsx:31]: indent your code

File: src/pages/Inputdemo/inputDemo.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+       alert('Form submitted Successfully');
+   }
+       
+ }
+ 
+  render() { 
+   
+    return (
+      <div>
+        <form style={formStyle}>
```
Comment (pinku4444) [inline src/pages/Inputdemo/inputDemo.jsx:39]: indent your code

File: src/pages/NoMatch/NoMatch.jsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+const NoMatch = () => {
+
+    return (
+
+        <div>
+            <center>
+                <h1 style={{color:'grey'}}> No page found! </h1>
```
Comment (pinku4444) [inline src/pages/NoMatch/NoMatch.jsx:9]: Remove inline css

File: src/pages/TextFieldDemo/TextFieldDemo.jsx [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+      this.setState({[nam]: val});
+   }
+
+   erroMsgHandler = (event) => {
+      let nam = event.target.name;
+      let val = event.target.value;
+      let err = '';
+      if(nam === 'name'){
+         if(val !== "" && Number(val)){
+            err = <strong style ={{color: 'red'}}> Please Enter A Valid Name</strong>
```
Comment (pinku4444) [inline src/pages/TextFieldDemo/TextFieldDemo.jsx:53]: Remove inline css

File: src/pages/TextFieldDemo/TextFieldDemo.jsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+      }
+      
+      this.setState({errmsg: err});
+      this.setState({[nam]: val});
+   }
+  render() { 
+   
+    return (
+      <div>
+        <form style={formStyle}>
```
Comment (pinku4444) [inline src/pages/TextFieldDemo/TextFieldDemo.jsx:65]: indent your code

File: src/pages/Trainee/Trainee.jsx [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+Modal.setAppElement('#root')
+
+
+export default class Trainee extends React.Component {
+    constructor(props){
+        super(props);
+           this.state = {
+                modalIsOpen: false
+        }
+        this.openModal = this.openModal.bind(this);
```
Comment (pinku4444) [inline src/pages/Trainee/Trainee.jsx:18]: You can use arrow function.

File: src/pages/Trainee/Trainee.jsx [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+            button: {
+                margin: theme.spacing(1),
+              },
+    }));
+        
+
+    render() {
+            const classes = this.useStyles;
+        return (
+            <div>
```
Comment (pinku4444) [inline src/pages/Trainee/Trainee.jsx:48]: indent your code

File: src/pages/Trainee/components/AddDialoge/AddDialoge.jsx [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+                margin: theme.spacing(1),
+              },
+    }));
+        
+
+    render() {
+        const classes = this.useStyles;
+        const { formData, submitted } = this.state;
+        return (
+            <div className={classes.container}>
```
Comment (pinku4444) [inline src/pages/Trainee/components/AddDialoge/AddDialoge.jsx:97]: indent your code

File: src/routes/AuthRoute.jsx [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+import LoginLayoutRoute from '../layouts/AuthLayout/AuthLayout';
+import Login from '../pages/login/Login';
+
+
+const AuthRoute = ()=> {
+    return(
+        <Router>
+        <div>
+        <Switch>
+          <Route exact path="/" component={Login}>
```
Comment (pinku4444) [inline src/routes/AuthRoute.jsx:13]: Why you are not using self-closing component ?

File: src/pages/login/Login.jsx [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import Link from '@material-ui/core/Link';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+// import Navbar from './NavBar';
```
Comment (pinku4444) [inline src/pages/login/Login.jsx:16]: Remove commented code.

File: src/components/TextField/TextField.jsx [hunk 16]
```diff
@@ -0,0 +1,17 @@
+// eslint-disable-next-line no-unused-vars
+import React, { Component } from 'react';
+class TextField extends Component {
+  
+  render() {
+    const { isError, ...rest } = this.props;
+    return (
+      <div>
```
Comment (pinku4444) [inline src/components/TextField/TextField.jsx:8]: indent your code

File: src/components/SelectField/SelectField.jsx [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+      default:
+        return <p></p>;
+    }
+  }
+
+    render(){
+      // eslint-disable-next-line no-unused-vars
+      const {  name, onChangeSelect, handleButton, isEnabled, ...rest} = this.props;
+      return (
+        <div>
```
Comment (pinku4444) [inline src/components/SelectField/SelectField.jsx:30]: indent your code

File: src/components/Math/Math.jsx [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+
+                <TextField name='second' placeholder='Second Number' 
+                onChange={this.handleInput} margin='dense' 
+                value={this.state.second}/><br/><br/>
+
+                <TextField name='operator' placeholder='Operator' 
+                onChange={this.handleInput} margin='dense' 
+                value={this.state.operator}/>
+                <br/>
+                <Select name='operator' onChange={this.handleInput}>
```
Comment (pinku4444) [inline src/components/Math/Math.jsx:78]: indent your code

File: src/App.jsx [hunk 19]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import NavBar from './layouts/components/Navbar/NavBar';
+import AuthLayout from './routes/AuthRoute';
+import PrivateRoute from './routes/PrivateRoute';
+
+function App() {
+  return (
+    <div>
```
Comment (pinku4444) [inline src/App.jsx:8]: indent your code

File: src/components/Math/Math.jsx (hunk 1 above)
Comment (varsha-22) [reply src/components/Math/Math.jsx:20]: okay Sir

File: src/components/RadioGroup/RadioGroups.jsx (hunk 2 above)
Comment (varsha-22) [reply src/components/RadioGroup/RadioGroups.jsx:8]: done i'll push the updated code and will let  you know.

File: src/images/index.js (hunk 3 above)
Comment (varsha-22) [reply src/images/index.js:2]: okay

File: src/layouts/components/Navbar/NavBar.jsx (hunk 6 above)
Comment (varsha-22) [reply src/layouts/components/Navbar/NavBar.jsx:31]: okay

File: src/pages/NoMatch/NoMatch.jsx (hunk 8 above)
Comment (varsha-22) [reply src/pages/NoMatch/NoMatch.jsx:9]: Some are left i ll remove them too.

File: src/pages/TextFieldDemo/TextFieldDemo.jsx (hunk 9 above)
Comment (varsha-22) [reply src/pages/TextFieldDemo/TextFieldDemo.jsx:53]: okay sir

File: src/pages/login/Login.jsx (hunk 15 above)
Comment (varsha-22) [reply src/pages/login/Login.jsx:16]: okay SIr

File: src/pages/Trainee/Trainee.jsx (hunk 11 above)
Comment (varsha-22) [reply src/pages/Trainee/Trainee.jsx:18]: okay will change that.

File: src/layouts/PrivateLayout/PrivateLayout.jsx (hunk 5 above)
Comment (varsha-22) [reply src/layouts/PrivateLayout/PrivateLayout.jsx:5]: okay Sir

**Pull request merged**